### PR TITLE
#1573 Add config option to disallow form content types in body listener

### PR DESCRIPTION
**Pull request by JornEngelbart**

### CODE REVIEW

File: EventListener/BodyListener.php [hunk 1]
```diff
@@ -41,17 +41,20 @@ class BodyListener
...
      * @param DecoderProviderInterface $decoderProvider
      * @param bool                     $throwExceptionOnUnsupportedContentType
+     * @param bool                     $acceptFormContentType
      * @param ArrayNormalizerInterface $arrayNormalizer
      * @param bool                     $normalizeForms
      */
     public function __construct(
         DecoderProviderInterface $decoderProvider,
         $throwExceptionOnUnsupportedContentType = false,
+        $acceptFormContentType = true,
```
Comment (Ener-Getick) [inline EventListener/BodyListener.php]: That's a bc break

**Comment by JornEngelbart** [comment]
I also fixed this for the 1.7 branch. Should I create another pull request for this?

**Comment by Ener-Getick** [comment]
Why don't you want to support forms?

**Comment by JornEngelbart** [comment]
@Ener-Getick Because there is a csrf vulnerability when doing a POST request to an endpoint that does not require any request body. When a user is already logged in a malicious form could cause the user to perform actions that he did not intent to do.

**Comment by Ener-Getick** [comment]
@JornEngelbart i'm not sure this belongs to this bundle as this part is managed by php and symfony themselves.
IMO you should create your own listener throwing an exception when having a form.

Wdyt @lsmith77 @xabbuh?

**Comment by JornEngelbart** [comment]
@Ener-Getick Seems inconsistent to me. There is an option to throw exceptions when a content-type is not supported. But at the same time two "special" content-types, namely the forms, are always supported.

**Comment by Ener-Getick** [comment]
Because forms are always supported (they are managed by php itself). 
What's the problem of having your own listener ? It's just a matter of a few lines.

**Comment by JornEngelbart** [comment]
@Ener-Getick Implementing an own listener is no problem. But IMO if you give an option to throw an exception on unsupported content-types it would be rather logical to be able to support or not support any content-type including forms (whether or not they are managed by PHP, these are content-types). Right?

**Comment by Ener-Getick** [comment]
I have no strong opinion about it, i'm just afraid it could bring more maintainance without bringing a real value imo.
I remembered [NelmioCorsBundle](https://github.com/nelmio/NelmioCorsBundle), why not using it ? It should be safer for csrf vulnérabilités.

**Comment by xabbuh** [comment]
Wouldn't it be better to then let the decoder provider decide whether or not support this content type and let the already existing exception mechanism step into?

**Comment by xabbuh** [comment]
I am closing here as the PR seems to be stalled. If someone wants to take over please create a new PR against the `2.x` branch.

**Pull request closed without merging**